packages/ui/src/components/Tooltip.tsx

```diff
@@ -0,0 +1,332 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useReducer,
+  useRef,
+  useState,
+} from "react";
+import {
+  ColorValue,
+  StyleProp,
+  StyleSheet,
+  View,
+  ViewStyle,
+} from "react-native";
+
+const styles = StyleSheet.create({
+  triangle: {
+    position: "absolute",
+    width: 0,
+    height: 0,
+    backgroundColor: "transparent",
+    borderStyle: "solid",
+    borderLeftWidth: 8,
+    borderRightWidth: 8,
+    borderBottomWidth: 16,
+    borderLeftColor: "transparent",
+    borderRightColor: "transparent",
+    borderBottomColor: "white",
+  },
+  container: {
+    position: "absolute",
+    width: "100%",
+    height: "100%",
+    left: 0,
+    top: 0,
+  },
+  tooltip: {
+    padding: "10px",
+    backgroundColor: "black",
+    opacity: 0.5,
+    zIndex: 10000,
+  },
+});
+
+type TriangleProps = {
+  pointerColor?: ColorValue;
+  style?: StyleProp<ViewStyle>;
+  placement: TooltipPosition;
+  trianglePosition: { top?: number; left?: number };
+};
+
+const triangleAnglesByPlacement: { [key: string]: number } = {
+  bottom: 0,
+  left: 90,
+  top: 180,
+  right: 270,
+};
+
+const Triangle: React.FC<TriangleProps> = ({
+  style,
+  pointerColor,
+  placement,
+  trianglePosition,
+}) => (
+  <View
+    style={StyleSheet.flatten([
+      styles.triangle,
+      {
+        borderBottomColor: pointerColor,
+        transform: [{ rotate: `${triangleAnglesByPlacement[placement]}deg` }],
+        ...trianglePosition,
+      },
+      style,
+    ])}
+  />
+);
+
+export type TooltipPosition =
+  | "top"
+  | "right"
+  | "bottom"
+  | "left"
+  | "overlay"
+  | "inset-bottom-left"
+  | "inset-bottom-right"
+  | "inset-top-left"
+  | "inset-top-right"
+  | "cursor";
+
+export type TooltipProps = {
+  placement?: TooltipPosition;
+  stayOnScreen?: boolean;
+  show: boolean;
+  anchorRef: React.MutableRefObject<View | null>;
+  children: React.ReactNode;
```

Comment:
Hope you don't mind: I modified this from being `text | children` to just being `children`, since `children` can _be_ a string in React (e.g. `<Foo>bar</Foo>`). We can make the union type work if you prefer, but it'll require some extra assertions in the component implementation to pass the type checks.

Reply:
Yeah, makes sense. I think I had in mind that the flexibility of passing children or a "simpler" `text=...` would be nice, but not tied to it.